packages/ra-core/src/dataProvider/useGetList.ts

```diff
@@ -66,13 +67,17 @@ export const useGetList = <RecordType extends RaRecord = any>(
     } = params;
     const dataProvider = useDataProvider();
     const queryClient = useQueryClient();
+    const { onError = noop, onSuccess = noop, ...queryOptions } = options;
```

Comment:
fun fact: useQuery v3 used to support an `onSettled` function, too. So we need to reimplement it here, too, using effects.